pkg/logs/tag/provider.go

```diff
@@ -62,7 +62,8 @@ func (p *provider) GetTags() []string {
 
 		// start timer if necessary
 		go func() {
-			<-time.After(p.expectedTagsDuration)
+			deadline := coreConfig.StartTime.Add(p.expectedTagsDuration)
+			<-time.After(time.Until(deadline))
```

Comment:
nit to initialize as much as possible in `NewProvider`: I'd move this logic to `NewProvider`, and in `NewProvider`: store the deadline under `expectedTagsDeadline` on the struct instead of storing the duration on the struct, and only set `submitExpectedTags = true` in `NewProvider` if the deadline is later than `time.Now()`.

Reply:
I understand why you say this as a detail of form, and I'll try to address it, but this makes the code a lot harder to unit test.